tests/test_test_tools/test_test_cases.py

```diff
@@ -369,3 +374,31 @@ def test_invoke(self):
         result = self.invoke("--version")
         assert result.exit_code == 0
         assert f"aea, version {aea.__version__}" in result.stdout
+
+
+class TestFlakyMany(AEATestCaseManyFlaky):
+    """Test that flaky tests are properly rerun."""
+
+    @pytest.mark.flaky(reruns=1)
+    def test_fail_on_first_run(self):
+        """Test failure on first run leads to second run."""
+        file = os.path.join(self.t, "test_file")
+        if self.run_count == 1:
+            open(file, "a").close()
+            raise AssertionError("Expected error to trigger rerun!")
+        assert self.run_count == 2, "Should only be rerun once!"
+        assert not os.path.isfile(file), "File should not exist"
+
```

Comment:
This shows that both the setup_class and teardown_class are called on each re-run.